addon/serializers/json-api-serializer.js

```diff
@@ -136,6 +136,13 @@ const JSONAPISerializer = JSONSerializer.extend({
     @private
   */
   _normalizeResourceHelper: function(resourceHash) {
+    if (Ember.isNone(resourceHash.type)) {
+      Ember.warn(this.warnMessageForUndefinedType(), false, {
```

Comment:
I wonder if Ember Data should be asserting here instead of a warning. @martndemus do you have a use case where you expect your type to be undefined?